src/legacy/ui/public/agg_types/agg_params.js

```diff
@@ -21,14 +21,17 @@ import '../filters/label';
 import { IndexedArray } from '../indexed_array';
 import { FieldParamType } from './param_types/field';
 import { OptionedParamType } from './param_types/optioned';
+import { SelectParamType } from './param_types/select';
 import { StringParamType } from './param_types/string';
 import { JsonParamType } from './param_types/json';
 import { BaseParamType } from './param_types/base';
 import { createLegacyClass } from '../utils/legacy_class';
 
 const paramTypeMap = {
   field: FieldParamType,
+  // todo: remove optioned type and use select instead;
   optioned: (OptionedParamType),
+  select: SelectParamType,
```

Comment:
All controls, which are using the `optioned` type should be rewritten to use `select` type since it has appropriate field names for the `EuiSelect` component.

Reply:
Since that is kind of part of this API I wonder if we could just keep the name `optioned` so far, and move all the existing options parameters in the same PR over to the new components?

Reply:
Discussed in slack.
Will be rewritten back to `optioned` when the last component with this type is EUIficated